[DOCS-4044] Update and move Reference Tables doc

maycmlee opened the pull request:
### What does this PR do?

- Adds Validation Rules section
- Updates the Reference Tables doc so it's not only for logs. 
- Moves docs, and images, to integrations guide folder.

### Motivation

DOCS-4044


### Additional Notes

Do not merge until validation rules functionality is rolled out (end Sept/early October).

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

dflorea reviewed (commented):
The changes look good. I would like to hold off on merging until the validation checks are available.

kayayarai commented on content/en/integrations/guide/reference-tables.md:
```diff
@@ -20,9 +21,22 @@ During the beta, there is a limit of 100 Reference Tables per account.
+| Names and headers can not be duplicated.											| Duplicated names are enumerated. For example example, if `fileid` is used twice as a name, the first instance becomes `fileid1` and the second instance becomes `fileid2`. If a name or header is enumerated and it exceeds the 56 characters, it is rejected and needs to be renamed. |
+| Names and headers cannot contain uppercase letters. 								| Names with uppercase letters are converted to lowercase. This conversion may result in duplicate names, which are then enumerated. For example, `Fileid` and `FileID` both become `fileid` and are enumerated to `fileid1` and `fileid2` respectively. |
+| Names and headers cannot contain spaces. 											| Spaces other than leading and trailing spaces are replaced with underscore `_` characters. Leading and trailing spaces are removed. For example, `customer names` is replaced with `customer_names`. |
+| Names and headers must start with a lowercase letter. 							| Uppercase characters are converted to lowercase. Other characters are removed. For example, `23Two_three` becomes `two_three`.	|
```
```suggestion
| Names and headers must start with a lowercase letter. 							| Uppercase characters are converted to lowercase. Non-letter leading characters are removed. For example, `23Two_three` becomes `two_three`.	|
```

kayayarai reviewed (changes requested):
Tiny text comment, but also, I think you want to add a `{{< nextlink` link to this in `/integrations/guide/_index.md`, and maybe remove the old one in `logs/guide/_index.md`. Or maybe leave that second one because it is closely related to logs? 🤷‍♀️

maycmlee commented:
> Tiny text comment, but also, I think you want to add a `{{< nextlink` link to this in `/integrations/guide/_index.md`, and maybe remove the old one in `logs/guide/_index.md`. Or maybe leave that second one because it is closely related to logs? 🤷‍♀️

Thanks @kayayarai! I forgot about the guide indexes. I removed it from the `logs/guide/_index.md` and added the link in /integrations/guide/_index.md`. I also added a space to all the `tag=<tag_name>` because it was aesthetically bothering me that there was no space between the title and tags: https://a.cl.ly/E0uZYvqQ

Could you re-review?
